vllm/model_executor/layers/linear.py

```diff
@@ -284,14 +286,20 @@ def __init__(self,
 
         if output_sizes is None:
             output_sizes = [output_size]
+
+        from vllm.model_executor.layers.quantization.compressed_tensors.compressed_tensors import (  # noqa: E501; TODO: temporary fix; have to fix circular import
+            CompressedTensorsLinearMethod)
+
         self.quant_method.create_weights(
             layer=self,
             input_size_per_partition=self.input_size,
             output_partition_sizes=self.output_partition_sizes,
             input_size=self.input_size,
             output_size=self.output_size,
             params_dtype=self.params_dtype,
-            weight_loader=self.weight_loader,
+            weight_loader=(self.weight_loader_v2 if isinstance(
+                self.quant_method, CompressedTensorsLinearMethod) else
+                           self.weight_loader),
```

Comment:
Per offline discussion with @robertgshaw2-neuralmagic, we may want to maintain a list of unsupported methods so that we could easily track when could we deprecate weight loader v1.

Reply:
@comaniac Updated.